twoliter/src/lock.rs

```diff
@@ -216,24 +191,16 @@ impl OCIArchive {
         self.cache_dir.join(self.digest.replace(':', "-"))
     }
 
-    async fn pull_image(&self) -> Result<()> {
+    async fn pull_image(&self, cli: ImageTool) -> Result<()> {
```

Comment:
readability nit
```suggestion
    async fn pull_image(&self, image_tool: ImageTool) -> Result<()> {
```

Although this has a name collision with the function (maybe `discover_image_tool`?)